scipy/stats/_survival.py

```diff
@@ -1,38 +1,202 @@
-from dataclasses import make_dataclass
+import warnings
+from dataclasses import dataclass, field
 import numpy as np
-from ._censored_data import CensoredData
+from scipy import special
+from scipy.stats._censored_data import CensoredData
+from scipy.stats._common import ConfidenceInterval
 
 
 __all__ = ['ecdf']
 
 
-ECDFResult = make_dataclass('ECDFResult', ['x', 'cdf', 'sf'])
+@dataclass
+class EmpiricalDistributionFunction:
+    """An empirical distribution function produced by `scipy.stats.ecdf`
+
+    Attributes
+    ----------
+    points : ndarray
+        The point estimate of the cumulative distribution function (CDF) or its
+        complement, the survival function (SF), at unique values of the sample.
+    """
+    points: np.ndarray
+    # Exclude these from __str__
+    _x: np.ndarray = field(repr=False)  # points at which function is estimated
+    _n: np.ndarray = field(repr=False)  # number "at risk"
+    _d: np.ndarray = field(repr=False)  # number of "deaths"
+    _sf: np.ndarray = field(repr=False)  # survival function for var estimate
+    _kind: str = field(repr=False)  # type of function: "cdf" or "sf"
+
+    def __init__(self, x, points, n, d, kind):
+        self.points = points
+        self._x = x
+        self._n = n
+        self._d = d
+        self._sf = points if kind == 'sf' else 1 - points
```

Comment:
This is private so ok. If we have some follow up consider naming this `self._complement` so it's not tainted with CDF/SF.